### ISSUE TITLE
Electrum v4 | TypeError("'set' object is not subscriptable") & ConnStringFormatError('Il nome host non si risolve (getaddrinfo fallito)')

### ISSUE THREAD

**Issue by theDavidCoen** [spesmilo/electrum#6291]
I'm trying to open my first LN channel with Electrum v4.
![Annotazione 2020-06-27 100623](https://user-images.githubusercontent.com/38695835/85919449-68f32c80-b86b-11ea-82e3-3455fac21bd0.png)
I get this kind of error when I enter my node ID (my personal node is this [https://1ml.com/node/0206ba1bccc9241fda43879095f54f276b41373ca236735c5357d55e385a8062e5])

![Annotazione 2020-06-27 100623 A](https://user-images.githubusercontent.com/38695835/85919444-61cc1e80-b86b-11ea-86f7-2077176b2884.png)

If I enter my node URI I get this error:

![Annotazione 2020-06-27 100623 B](https://user-images.githubusercontent.com/38695835/85919338-8a074d80-b86a-11ea-839f-998a033e7087.png)

Could it be due to the fact that my personal node runs with Tor?

**Comment by ecdsa** [comment]
can you enable logging, and paste the part of the log that is around that error?

**Comment by theDavidCoen** [comment]
> can you enable logging, and paste the part of the log that is around that error?

Can you guide me on this? I used Electrum as a watch-only wallet, until now

**Comment by ecdsa** [comment]
enable 'write to logfile' in your preferences, trigger the error and inspect the logfile

**Comment by theDavidCoen** [comment]
Got it, but I have only this info:
20200627T101147.912075Z |     INFO | logging | Electrum version: 4.0.0b0 - https://electrum.org - https://github.com/spesmilo/electrum
20200627T101147.912075Z |     INFO | logging | Python version: 3.7.7 (tags/v3.7.7:d7c567b08f, Mar 10 2020, 09:44:33) [MSC v.1900 32 bit (Intel)]. On platform: Windows-10-10.0.18362-SP0
20200627T101147.912075Z |     INFO | logging | Logging to file: C:\Users\navig\AppData\Roaming\Electrum\logs\electrum_log_20200627T101147Z_13708.log
20200627T101147.912075Z |     INFO | logging | Log filters: verbosity '', verbosity_shortcuts ''

**Comment by theDavidCoen** [comment]
I tried to open the channel many times, but the logs still the same.

**Comment by ecdsa** [comment]
a new logfile is created for every session, and it should have more than 4 lines

**Comment by theDavidCoen** [comment]
I think I got it.
I started Tor Browser, went to Electrum, Network, Proxy, selected "Use Tor proxy at port 9150" and opened a channel to my node using the URI (with @ onion).
I'm waiting for the 3 confirmation now.

I hope it could be helpful for other people!

**Comment by SomberNight** [comment]
probably duplicate of https://github.com/spesmilo/electrum/issues/6274, fixed in 4.0.0b1
The exception is different due to the different python versions used.

```
> python3.7
Python 3.7.7 (tags/v3.7.7:d7c567b08f, Mar 10 2020, 10:41:24) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> a = set([1,2,3])
>>> import random
>>> random.choice(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "...\Python37\lib\random.py", line 262, in choice
    return seq[i]
TypeError: 'set' object is not subscriptable

> python3.6
Python 3.6.8 (tags/v3.6.8:3c6b436a57, Dec 24 2018, 00:16:47) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> a = set([1,2,3])
>>> import random
>>> random.choice(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "...\Python36\lib\random.py", line 261, in choice
    return seq[i]
TypeError: 'set' object does not support indexing
```

The getaddrinfo failed (when using the onion URI, without a Tor proxy, is expected (although maybe not user-friendly).
If you want, you can open a new issue specifically for this part.
